ecs-agent/tmds/handlers/fault/v1/types/types.go

```diff
@@ -65,13 +70,29 @@ func (request NetworkBlackholePortRequest) ValidateRequest() error {
 		return fmt.Errorf(invalidValueError, *request.Protocol, "Protocol")
 	}
 
-	if *request.TrafficType != "ingress" && *request.TrafficType != "egress" {
+	if *request.TrafficType != TrafficTypeIngress && *request.TrafficType != TrafficTypeEgress {
 		return fmt.Errorf(invalidValueError, *request.TrafficType, "TrafficType")
 	}
+	if err := validateNetworkFaultRequestSources(request.SourcesToFilter, "SourcesToFilter"); err != nil {
+		return err
+	}
 
 	return nil
 }
 
+// Adds a source to SourcesToFilter
+func (request *NetworkBlackholePortRequest) AddSourceToFilterIfNotAlready(source string) {
+	if request.SourcesToFilter == nil {
+		request.SourcesToFilter = []*string{}
+	}
+	for _, src := range request.SourcesToFilter {
+		if src != nil && *src == source {
```

Comment:
non-blocking: we can use aws.StringValue(...) which can help us check the nil

Reply:
Good catch! Let me fix that as a follow-up.